src/titiler/core/titiler/core/factory.py

```diff
@@ -812,6 +824,299 @@ def wmts(
+                            detail="bbox needs 4 coordinates separated by commas",
+                        )
+                elif key == "version" and value not in set(WMSVersion):
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Version specified not allowed. Allowed versions include: {[w.value for w in list(WMSVersion)]}",
+                    )
+                elif key == 'format':
+                    value = WMSMediaType(value)
+                    
+                if key == "layers":
+                    out_wms_params[key] += [value]
+                else:
+                    out_wms_params[key] = value
+                    
+            if out_wms_params.get("bbox"):
+
+                # WMS 1.3.0 is lame and flips the coords of EPSG:4326
+                if (
+                    out_wms_params.get("version") == "1.3.0"
+                    and out_wms_params.get("srs")
+                    and out_wms_params.get("srs") == "EPSG:4326"
+                ):
+                    out_wms_params["bbox"] = [
+                        out_wms_params["bbox"][1],
+                        out_wms_params["bbox"][0],
+                        out_wms_params["bbox"][3],
+                        out_wms_params["bbox"][2],
+                    ]
+                    
+                # Overwrite CRS:84 with EPSG:4326 when specified
+                elif (
+                    out_wms_params.get("version") == "1.3.0"
+                    and out_wms_params.get("srs") == "CRS:84"
+                ):
+                    out_wms_params["srs"] = "EPSG:4326"
+                    
+                # `bounds` is compatible with reading from rio.io.reader
+                out_wms_params['bounds'] = out_wms_params['bbox']
+                    
+            print(out_wms_params)
+            return out_wms_params
+
+        @self.router.get("/wms")
+        def wms(
+            _request: Request,
+            request: WMSRequestType = "GetCapabilities",
+            version: WMSVersion = "1.1.1",
+            format: WMSMediaType = Query(
+                default=None, 
+                description="Output image type. Default is auto."
+            ),
+            bbox: Union[str, None] = Query(
+                default=None,
+                description="Bounding box for the WMS query. Should be in format of 'minx,miny,maxx,maxy'.",
+            ),
+            width: Union[int, None] = Query( 
+                default=None,
+                description="Width (in pixels) of returned GetMap image."
+            ),
+            height: Union[int, None]  = Query( 
+                default=None,
+                description="Height (in pixels) of returned GetMap image."
+            ),
+            srs: Union[EPSGCodes, None] = Query(
+                default=None,
+                description="Coordinate system of bbox as an EPSG code",
+            ),
```

Comment:
I think we could allow `string` as input, it will be up to the user to gives the right one. rasterio should be able to parse it (either if it's a string or an uri)